keanu-project/src/test/java/io/improbable/keanu/vertices/generic/nonprobabilistic/PrintVertexTest.java

```diff
@@ -0,0 +1,196 @@
+package io.improbable.keanu.vertices.generic.nonprobabilistic;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import io.improbable.keanu.algorithms.mcmc.MetropolisHastings;
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.tensor.dbl.DoubleTensor;
+import io.improbable.keanu.tensor.intgr.IntegerTensor;
+import io.improbable.keanu.vertices.NonSaveableVertex;
+import io.improbable.keanu.vertices.Vertex;
+import io.improbable.keanu.vertices.bool.BooleanVertex;
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.probabilistic.UniformVertex;
+import io.improbable.keanu.vertices.generic.nonprobabilistic.operators.unary.UnaryOpVertex;
+import io.improbable.keanu.vertices.intgr.nonprobabilistic.ConstantIntegerVertex;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.PrintStream;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.core.IsInstanceOf.instanceOf;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PrintVertexTest {
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @Mock
+    PrintStream printStream;
+
+    @Before
+    public void setup() {
+        PrintVertex.setPrintStream(printStream);
+    }
+
+    @Test
+    public void whenCreatedThenParentIsSet() {
+        final Vertex<DoubleTensor> parent = new ConstantDoubleVertex(30);
+
+        final PrintVertex<DoubleTensor> sut = new PrintVertex<>(parent);
+
+        assertThat(sut.getParents(), equalTo(ImmutableSet.of(parent)));
+    }
+
+    @Test
+    public void whenCreatedWithNullParentThenFails() {
+        thrown.expect(NullPointerException.class);
+        new PrintVertex<>(null);
+    }
+
+    @Test
+    public void whenCreatedWithNullPrefixParentThenFails() {
+        thrown.expect(NullPointerException.class);
+        final DoubleVertex parent = new ConstantDoubleVertex(30);
+        new PrintVertex<>(parent, null, false);
+    }
+
+    @Test
+    public void whenPrintStreamIsNullThenFails() {
+        thrown.expect(NullPointerException.class);
+        PrintVertex.setPrintStream(null);
+    }
+
+    @Test
+    public void whenOnlyParentIsSpecifiedThenDefaultPresentationIsUsed() {
+        final Vertex<DoubleTensor> parent = new ConstantDoubleVertex(30);
```

Comment:
This is a good place to use a mock. If the implementation of `ConstantDoubleVertex.toString()` changed (which is quite likely) then this test would fail, but we don't actually care about `ConstantDoubleVertex` in this test. It's enough to mock a `Vertex` and stub its `toString` method.